draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -385,128 +355,160 @@ plus a string suitable for computing a digest, but such approaches can easily le
 an application that needs to access the disclosed claim values. The
-HS-Disclosures object is only intended to be used by a Verifier to check
+Disclosures are only intended to be used by a Verifier to check
 the digests over the source strings and to extract the original JSON
 data. The original JSON data is then used by the application. See
-(#processing_model) for details.
+(#verifier_verification) for details.
 
 ## Format of an SD-JWT
 
 An SD-JWT is a JWT that MUST be signed using the Issuer's private key. The
-payload of an SD-JWT MUST contain the `sd_digests` and `sd_digest_derivation_alg` claims
-described in the following, and MAY contain a Holder's public key or a reference
+payload of an SD-JWT MUST contain the `sd_digest_derivation_alg` claim
+described in the following, MAY contain one or more selectively disclosable claims, and MAY contain a Holder's public key or a reference
 thereto, as well as further claims such as `iss`, `iat`, etc. as defined or
 required by the application using SD-JWTs.
 
-### `sd_digests` Claim (Digests of Selectively Disclosable Claims) {#sd_digests_claim}
-
-The property `sd_digests` MUST be used by the Issuer to include digests of the salted claim values for any claim that is intended to be selectively disclosable.
+### Selectively Disclosable Claims
 
-The Issuer MUST choose a new, cryptographically random salt value for
-each claim value. The salt value MUST then be encoded as a string. It is
-RECOMMENDED to base64url-encode the salt value.
+For each claim that is to be selectively disclosed, the Issuer creates a Disclosure object, hashes it, and includes the hash instead of the original claim in the SD-JWT, as described next. The Disclosures are then sent to the Holder.
 
-The Issuer MUST generate the digests over a JSON literal according to
-[@!RFC8259] that is formed by
-JSON-encoding an object with the following contents:
+#### Creating Disclosures {#creating_disclosures}
+The Issuer MUST create a Disclosure object for each selectively disclosable claim as follows:
 
- * REQUIRED with the key `s`: the salt value,
- * REQUIRED with the key `v`: the claim value (either a string or a more complex object, e.g., for the [@OIDC] `address` claim),
- * OPTIONAL, with the key `n`: the claim name (if Claim Name Blinding is to be used for this claim).
+ * Create an array of three elements in this order:
+   1. A salt value. See (#salt-entropy) and (#salt_minlength) for security considerations. The salt value MUST be unique for each claim that is to be selectively disclosed. It is RECOMMENDED to base64url-encode the salt value, producing a string. Any other type that is allowed in JSON MAY be used, e.g., a number.
+   2. The claim name, or key, as it would be used in a regular JWT. This MUST be a string.
+   3. The claim's value, as it would be used in a regular JWT. The value MAY be of any type that is allowed in JSON, including numbers, strings, booleans, arrays, and objects.
```

Comment:
```suggestion
   3. The claim's value, as it would be used in a regular JWT body. The value MAY be of any type that is allowed in JSON, including numbers, strings, booleans, arrays, and objects.
```